src/cdk-experimental/selection/selection.spec.ts

```diff
@@ -0,0 +1,691 @@
+         expect(component.getRow(0).getAttribute('aria-selected')).toBe('false');
+         expect(component.getRow(1).getAttribute('aria-selected')).toBe('false');
+         expect(component.getRow(2).getAttribute('aria-selected')).toBe('false');
+         expect(component.getRow(3).getAttribute('aria-selected')).toBe('false');
+
+         component.clickSelectionToggle(0);
+
+         expect(component.getRow(0).getAttribute('aria-selected')).toBe('true');
+
+         component.clickSelectionToggle(0);
+
+         expect(component.getRow(0).getAttribute('aria-selected')).toBe('false');
+       }));
+  });
+});
+
+describe('cdkSelectionColumn with multiple = false', () => {
+  let fixture: ComponentFixture<SingleSelectionColumnApp>;
+  let component: SingleSelectionColumnApp;
+
+  beforeEach(async(() => {
+    TestBed
+        .configureTestingModule({
+          imports: [
+            CdkSelectionModule,
+            CdkTableModule,
+          ],
+          declarations: [
+            SingleSelectionColumnApp,
+          ]
+        })
+        .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SingleSelectionColumnApp);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should not show select all', () => {
+    expect(component.elementRef.nativeElement.querySelector('input[cdkselectall]')).toBe(null);
+  });
+
+  it('should allow selecting one single row', fakeAsync(() => {
+       expect(component.getSelectionToggle(0).checked).toBe(false);
+       expect(component.getSelectionToggle(1).checked).toBe(false);
+       expect(component.getSelectionToggle(2).checked).toBe(false);
+       expect(component.getSelectionToggle(3).checked).toBe(false);
+
+       component.clickSelectionToggle(0);
+
+       expect(component.getSelectionToggle(0).checked).toBe(true);
+
+       component.clickSelectionToggle(1);
+
+       expect(component.getSelectionToggle(0).checked).toBe(false);
+       expect(component.getSelectionToggle(1).checked).toBe(true);
+
+       component.clickSelectionToggle(1);
+       expect(component.getSelectionToggle(1).checked).toBe(false);
+     }));
+});
+
+@Component({
+  template: `
+    <ul cdkSelection [dataSource]="data" [cdkSelectionMultiple]="true"
+        (cdkSelectionChange)="selectionChange = $event">
+      <button cdkSelectAll #toggleAll="cdkSelectAll" (click)="toggleAll.toggle($event)">
+        {{selectAllState(toggleAll.indeterminate | async, toggleAll.checked | async)}}
+      </button>
+      <li *ngFor="let item of data; index as i">
+        <button cdkSelectionToggle #toggle="cdkSelectionToggle"
+            [cdkSelectionToggleValue]="item"
+            [cdkSelectionToggleIndex]="i"
+            (click)="toggle.toggle()">
+          {{(toggle.checked | async) ? 'checked' : 'unchecked'}}
+        </button>
+        {{item}}
+      </li>
+    </ul>`
+})
+class MultiSelectionApp {
```

Comment:
nit: could you need these test components along the pattern `ListWithMultiSelection`, `ListWithSingleSelection`, `MultiSelectTableWithSelectionColumn`, etc.